modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -2044,7 +2044,11 @@ private boolean recordable(TcpDiscoveryAbstractMessage msg) {
      * @param rmtPerms The second set of permissions.
      * @return {@code True} if given parameters contain the same permissions, {@code False} otherwise.
      */
-    private boolean permissionsEqual(SecurityPermissionSet locPerms, SecurityPermissionSet rmtPerms) {
+    private boolean permissionsEqual(@Nullable SecurityPermissionSet locPerms,
+        @Nullable SecurityPermissionSet rmtPerms) {
+        if (locPerms == null || rmtPerms == null)
```

Comment:
What if both `locPerms` and `rmtPerms` are `null`?

Reply:
Both nulls means something wrong with configuration because it shouldn't be possible, because Ignite checks configuration consistency.

`null` `rmtPerms` means authenticator on current node doesn't authenticated node.
`null` `locPerms` means authenticator on joining node doesn't authenticated node. In this case node shouldn't start because of `null` in `localAuthentication`.